Admin fixup

jameshiew opened the pull request:
Use ModelAdmin classes, also mark many fields with `blank=True` so that objects with nullable fields can be saved in the admin panel without having to specify a value.

I've tried to keep changes minimal and didn't mark all fields as readonly as I don't know how people are currently using the admin - it is useful for making quick changes to local objects during development, as long as you're aware changes don't sync up to Stripe automatically (apart from for `Plan` objects).

Re #385

jameshiew committed:
Use admin.ModelAdmin classes

Make stripe_id, livemode, stripe_timestamp fields readonly and always
place at the top of admin pages

Use date_hierarchy based on stripe_timestamp where possible

Explicitly set 'fields' attribute on ModelAdmin classes

Don't populate >1 Subscription inline for Customers by default

jameshiew committed:
Mark various fields with blank=True

This allows changes to be made to local objects via the admin panel

kavdev reviewed (changes requested):
Go ahead and mark all fields as readonly.

Also, migrations are now out of date.

jameshiew committed:
Remake migration

jameshiew committed:
Typo: data -> date

jameshiew committed:
Make all Stripe object fields readonly

Also disable addition of new Stripe objects via the admin panel

Make SubscriptionInline readonly

Make InvoiceItemInline readonly

Remove unneeded CustomerAdmin method

kavdev commented on djstripe/urls.py:
```diff
@@ -26,7 +26,7 @@
 from . import views
 
-
+app_name = 'djstripe'
```
Will this affect anyone's current code? i.e. do I have to add it to the backwards-incompatible section of the changelog?

IIRC this makes these urls namespaced if included (It's been a while since I've used ``app_name``)

jameshiew replied:
Not sure, adding this lets you write `url(r'^stripe/', include('djstripe.urls'))` instead of `url(r'^stripe/', include('djstripe.urls', namespace="djstripe"))` but I meant to separate this commit out from this admin stuff. I'll look into it and do it as a separate PR

Commit:
Add `statement_descriptor` to Plan admin readonly fields

Only a Plan's name can be changed and pushed to Stripe

kavdev commented on djstripe/migrations/0022_auto_20170121_1637.py:
```diff
@@ -0,0 +1,153 @@
+# -*- coding: utf-8 -*-
```
The explicit bytestrings will cause python 3 users to need another migration; please replace them.

kavdev reviewed (changes requested):
One last fix with the migration and it's good to merge!
